virtcontainers/sandbox.go

```diff
@@ -561,6 +566,23 @@ func (s *Sandbox) Status() SandboxStatus {
 	}
 }
 
+// Monitor returns a error channel for watcher to watch at
+func (s *Sandbox) Monitor() (chan error, error) {
+	if s.state.State != StateRunning {
+		return nil, fmt.Errorf("Sandbox not running")
+	}
+
+	if s.monitor == nil {
+		s.Lock()
+		if s.monitor == nil {
```

Comment:
Second time you check if `s.monitor` is nil, I think that's a bit too much :smiley:

Reply:
It ensures we only create and set the monitor once. Otherwise concurrent callers might race in and we end up setting monitor twice.

Reply:
You can simply do the lock and test the monitor value. This is really cheap here.

Reply:
Fair enough, I'll change it.

Reply:
Done.